packages/nimble-components/src/table/types.ts

```diff
@@ -123,6 +136,12 @@ export interface TableRowSelectionEventDetail {
     selectedRecordIds: string[];
 }
 
+export interface TableRowExpandToggleEventDetail {
+    oldState: boolean;
+    newState: boolean;
+    recordId?: string;
```

Comment:
`recordId` doesn't need to be optional, does it? Shouldn't any expandable row have a defined record id?

Reply:
This has been updated as part of [this PR](https://github.com/ni/nimble/pull/1679).